src/components/core/Reloader.react.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux'
+import {getReloadHash} from "../../actions/api";
+
+class Reloader extends React.Component {
+    constructor(props) {
+        super(props);
+        if (props.config.hot_reload) {
+            const { interval } = props.config.hot_reload;
+            this.state = {
+                hash: null,
+                interval,
+                disabled: false
+            }
+        } else {
+            this.state = {
+                disabled: true
+            }
+        }
+        this._intervalId = null;
+    }
+
+    componentDidUpdate() {
+        const {reloadHash, dispatch} = this.props;
+        if (reloadHash.status === 200) {
+            if (this.state.hash === null) {
+                this.setState({hash: reloadHash.content.reloadHash});
+                return;
+            }
+            if (reloadHash.content.reloadHash !== this.state.hash) {
+                // eslint-disable-next-line no-undef
+                window.clearInterval(this._intervalId);
```

Comment:
So the `no-undef` rule here should not trigger if it's sure that `this._intervalId` is set, right? Maybe it's better to write a check here that makes sure that it's not `null`, instead of disabling the rule.

Reply:
The `no-undef` is for `window`, it highlight as error in pycharm.

Reply:
Ah ok! Hmm maybe that rule can be turned off globally then, if it's always going to trigger on `window`?

Reply:
Last question: isn't it maybe better to put the `rootReducer` in a separate file, and call it something else, like `reloadReducer` or whatever?

Reply:
> maybe that rule can be turned off globally then

I think we can put `window` in a global exclude or something. Anyway it underline as error in IDE, I just hit `alt-enter` and select a fix, doesn't take more than a second.

> Last question: isn't it maybe better to put the rootReducer in a separate file

It's only a couple lines and is the last reducer exported, keeping it in `reducer` make more sense to me.